App/Sagas/Migration.ts

```diff
@@ -32,6 +44,71 @@ export function * migrate(dispatch: Dispatch) {
   yield call(letScreenSleep)
 }
 
+// Can be run on each node online
+export function * runRecurringMigrationTasks () {
+  const announcement = yield select(getAnnouncement)
+  if (announcement) {
+    const {peerId, address, username, previous} = announcement
+    try {
+      yield call(announceMigration, peerId, address, username, previous)
+    } finally {
+      // If no error, mark as successful
+      yield put(MigrationActions.announceSuccess())
+    }
+  }
+  const peers = yield select(getNetwork)
+  for (const peer of peers) {
+    try {
+      // for each contact ask if they've migrated
+      const contact = yield call(findContact, peer)
+      if (contact) {
+        yield call(addContact, contact.peerId, contact.address, contact.username)
+      } else {
+        throw new Error('peer not found')
+      }
+    } finally {
+      // If no error, mark as successful
+      yield put(MigrationActions.connectionSuccess(peer))
+    }
+  }
+}
+
+// Should be run the first time
+export function * migrateConnections() {
+  // announce to other peers
+  const previousItems: PeerIdItem[] = yield call(getItems, PREVIOUS_ID_PATH)
+  if (previousItems.length) {
+    const previous = previousItems[0]
+    const peerId = yield select(getPeerId)
+    const address = yield select(getAddress)
+    const username = yield select(getUsername) || ''
+    yield put(MigrationActions.announceMigration(peerId, address, username, previous.peerid))
+  }
+  // locate old peers
+  const threadItems: ThreadItem[] = yield call(getItems, THREADS_FILE_PATH)
+  const peers = [...new Set(([] as string[]).concat(...threadItems.map((thread) => thread.peers)))]
+  yield put(MigrationActions.connectToPeers(peers))
+  // run it for the first time
+  yield call(runRecurringMigrationTasks)
```

Comment:
run it the first time here... could also just wait for nodeOnline to run it the first time actually